specs/FamilySafety.md

```diff
@@ -0,0 +1,114 @@
+    // FamilySafetyAllowedUris to show app content and still honor general filter settings set
+    // by parents.
+    CoreWebView2EnvironmentOptions options = new CoreWebView2EnvironmentOptions();
+    options.IsFamilySafetyEnabled = true;
+    // appassets.example is used as an example app content. All subdomain are added to the allow list
+    options.FamilySafetyAllowedUris.Add("https://appassets.example/*");
+    auto environment = await CoreWebView2Environment.CreateAsync(BrowserExecutableFolder, UserDataFolder, options);
+}
+```
+## Win32 C++
+```cpp
+HRESULT InitializeWebView()
+{
+    // If parents set filtering rules to only allowed sites. App developers can use
+    // FamilySafetyAllowedUris to show app content and still honor general filter settings set
+    // by parents.
+    auto options = Microsoft::WRL::Make<CoreWebView2EnvironmentOptions>();
+    options->put_IsFamilySafetyEnabled(TRUE);
+    // appassets.example is used as an example app content. All subdomain are added to the allow list
+    const WCHAR* allowedUris[1] = {L"https://appassets.example/*"};
+    options->SetFamilySafetyAllowedUris(1, allowedUris);
+    HRESULT hr = CreateCoreWebView2EnvironmentWithOptions("", nullptr, options.Get(),
+    Callback<ICoreWebView2CreateCoreWebView2EnvironmentCompletedHandler>(
+        this, &AppWindow::OnCreateEnvironmentCompleted)
+        .Get());
+}
+```
+
+# API Details    
+```
+interface ICoreWebView2EnvironmentOptions3;
+
+/// Additional options used to create WebView2 Environment.
+[uuid(D0965AC5-11EB-4A49-AA1A-C8E9898F80AF), object, pointer_default(unique)]
+interface ICoreWebView2EnvironmentOptions3 : IUnknown {
+  /// `IsFamilySafetyEnabled` property is to enable/disable family safety feature.
+  /// It is `FALSE` by default.
+  /// When `IsFamilySafetyEnabled` is `TRUE` WebView2 provides the same Family Safety 
+  /// functionality as the Edge browser. 
+  /// Family Safety is a set of features available on Windows for managing children Internet 
+  /// priviliges. Microsoft account may be linked to have a family relationship, with adults 
+  /// and children. Adults have certain managements options over the children in their family, 
+  /// where each option is applied per-child. Edge browser provide the following options: 
+  /// Activity reporting, Web filtering and SafeSearch. Please see 
+  /// https://aka.ms/EdgeFamilySafetyFeatureOverview for more details on each feature.
+  [propget] HRESULT IsFamilySafetyEnabled([out, retval] BOOL* value);
+  /// Sets the `IsFamilySafetyEnabled` property.
+  [propput] HRESULT IsFamilySaFetyEnabled([in] BOOL value);
+
+  /// `GetFamilySafetyAllowedUris` and `SetFamilySafetyAllowedUris` allow developers to get and set
+  /// the list of URIs that will be allowed by the Family Safety filter even when in Block All mode.
+  /// Family Safety provides web filtering control in two modes: Allow-all and Block-all. In 
```

Comment:
How do the allow-all and block-all modes relate to this API? Is block-all another way of saying IsFamilySafetyEnabled==true?

Reply:
Here's my interpretation:

| IsFamilySafetyEnabled | Block mode | Result |
|-|-|-|
| False | * | All URIs are allowed |
| True | Allow-all | Parent provides list of block URIs; anything unknown is allowed |
| True | Block-all | Parent provides list of allowed URIs; anything unknown is blocked |

Reply:
This is a good table we should include in our documentation:

```
| IsFamilySafetyEnabled | Family Safety Default Block Mode | Result |
| --- | --- | --- |
| `False` | * | All URIs are allowed |
| `True` | Allow by default | Parent provides list of block URIs; anything unknown is allowed | 
| `True` | Block by default | Parent provides list of allowed URIs; App provides FamilySafetyAdditionalAllowedUris list of additional allowed URIs; anything unknown is blocked |
```

Also, please ensure this documentation to describe in words what the table says esp including how FamilySafetyAdditionalAllowedUris interacts with it.